components/odh-notebook-controller/go.mod

```diff
@@ -78,3 +79,5 @@ require (
 )
 
 replace google.golang.org/grpc => google.golang.org/grpc v1.56.3
+
+replace github.com/kubeflow/kubeflow/components/notebook-controller => ../notebook-controller
```

Comment:
if we don't want to do this, I can put the test directly under components/ as a separate component; but I'd still like to do it this way, so if there's no opposition to the general direction, I'd like to go test it out in cpaas